pkg/storage/postgres/postgres.go

```diff
@@ -261,8 +261,8 @@ func putRecord(ctx context.Context, q querier, record *databroker.Record) error
 	} else {
 		_, err = q.Exec(ctx, `
 			DELETE FROM `+schemaName+`.`+recordsTableName+`
-			WHERE type=$1 AND id=$2 AND version<$3
```

Comment:
checking the version here was unnecessary because now the updates are serialized, so a row won't get updated concurrently